cmd/ctr/commands/run/run_unix.go

```diff
@@ -117,15 +124,10 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 	if context.IsSet("gpus") {
 		opts = append(opts, nvidia.WithGPUs(nvidia.WithDevices(context.Int("gpus")), nvidia.WithAllCapabilities))
 	}
-	if context.IsSet("config") {
-		var s specs.Spec
-		if err := loadSpec(context.String("config"), &s); err != nil {
-			return nil, err
-		}
-		spec = containerd.WithSpec(&s, opts...)
-	} else {
-		spec = containerd.WithNewSpec(opts...)
-	}
+
+	var s specs.Spec
+	spec = containerd.WithSpec(&s, opts...)
```

Comment:
Perhaps `containerd.WithSpec` could be made to tolerate a `nil` for the first argument (in which case it initialises the spec ptr internally with a zero val struct). Saves an, incredibly minor, bit of faff for the caller in these cases.

Reply:
I did explore this, but you'd have to pass in a pointer to the pointer to make this work with `nil`.

Reply:
I misread `spec` as being an actual `Spec`, but of course its a `Opt` thing which consumes `s` in a deferred way. Sorry for the noise.